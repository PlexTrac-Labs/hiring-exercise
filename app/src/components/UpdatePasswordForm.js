import React, { useRef, useState, useEffect } from "react";
import axios from "axios";

import CredentialWarning from "./CredentialWarning";
export default function UpdatePasswordForm({ user }) {
  const currentPasswordRef = useRef();
  const newPasswordRef = useRef();
  const retypePasswordRef = useRef();

  const [isWrongLogin, setIsWrongLogin] = useState(false);

  function handlePasswordSubmit(e) {
    e.preventDefault();
    const currentPassword = currentPasswordRef.current.value;
    const newPassword = newPasswordRef.current.value;
    const retypePassword = retypePasswordRef.current.value;
    if (newPassword !== retypePassword) {
      return;
    }
    axios
      .put(`/password/${user._id}`, {
        password: currentPassword,
        newPassword: newPassword
      })
      .then(res => {
        window.location = "/profile";
      })
      .catch(error => {
        setIsWrongLogin(true);
      });
  }

  return (
    <form className="updatePassword" onSubmit={handlePasswordSubmit}>
      <div>
        <input
          id="currentPasswordInput"
          ref={currentPasswordRef}
          type="text"
          name="currentPassword"
          required
        />
        <label htmlFor="currentPasswordInput">Current Password</label>
      </div>
      <div>
        <input
          id="newPasswordInput"
          ref={newPasswordRef}
          type="text"
          name="newPassword"
          required
        />
        <label htmlFor="newPasswordInput">New Password</label>
      </div>
      <div>
        <input
          id="newPasswordInput"
          ref={retypePasswordRef}
          type="text"
          name="retypePassword"
          required
        />
        <label htmlFor="newPasswordInput">Retype New Password</label>
      </div>
      <button className="w-100 btn btn-lg btn-primary" type="submit">
        Submit
      </button>
      {isWrongLogin ? <CredentialWarning /> : null}
    </form>
  );
}
