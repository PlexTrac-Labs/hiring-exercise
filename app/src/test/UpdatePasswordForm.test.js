import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import UpdatePasswordForm from "../components/UpdatePasswordForm";

//Test to make sure the App, entry point to everything works.
it("renders correctly", async () => {
  const tree = renderer.create(<UpdatePasswordForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
