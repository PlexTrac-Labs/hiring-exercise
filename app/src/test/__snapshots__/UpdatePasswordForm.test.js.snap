// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="updatePassword"
  onSubmit={[Function]}
>
  <div>
    <input
      id="currentPasswordInput"
      name="currentPassword"
      required={true}
      type="text"
    />
    <label
      htmlFor="currentPasswordInput"
    >
      Current Password
    </label>
  </div>
  <div>
    <input
      id="newPasswordInput"
      name="newPassword"
      required={true}
      type="text"
    />
    <label
      htmlFor="newPasswordInput"
    >
      New Password
    </label>
  </div>
  <div>
    <input
      id="newPasswordInput"
      name="retypePassword"
      required={true}
      type="text"
    />
    <label
      htmlFor="newPasswordInput"
    >
      Retype New Password
    </label>
  </div>
  <button
    className="w-100 btn btn-lg btn-primary"
    type="submit"
  >
    Submit
  </button>
</form>
`;
